cluster/swarm/cluster.go

```diff
@@ -452,9 +452,7 @@ func (c *Cluster) Image(IDOrName string) *cluster.Image {
 		return nil
 	}
 
-	c.RLock()
-	defer c.RUnlock()
-	for _, e := range c.engines {
```

Comment:
As long as operation at `Engine` level is mutual exclusive, this should be fine. `e.Image` may or may not find image from a node if the node may go away. But that should be ok. In distributed systems, finding the image at this moment doesn't guarantee finding the image next. System design needs to take that into consideration.